static/js/recent_topics.js

```diff
@@ -68,10 +81,88 @@ exports.process_topic_edit = function (old_stream_id, old_topic, new_topic, new_
     exports.process_messages(new_topic_msgs);
 };
 
-exports.launch = function () {
-    const rendered_body = render_recent_topics_body();
+function format_topic(topic_data) {
+    const last_msg = message_store.get(topic_data.last_msg_id);
+    if (last_msg === undefined) {
+        return false;
+    }
+    const stream = last_msg.stream;
+    const stream_id = last_msg.stream_id;
+    const topic = last_msg.topic;
+    const time = new XDate(last_msg.timestamp * 1000);
+    const last_msg_time = timerender.last_seen_status_from_date(time);
+    return {
+        stream_id: stream_id,
+        stream: stream,
+        topic: topic,
+        unread_count: unread.unread_topic_counter.get(stream_id, topic),
+        last_msg_time: last_msg_time,
+        stream_url: hash_util.by_stream_uri(stream_id),
+        topic_url: hash_util.by_stream_topic_uri(stream_id, topic),
+    }
+}
+
+function format_all_topics() {
+    const topics_array = [];
+    for (const [, value] of exports.get()) {
+        topics_array.push(format_topic(value));
+    };
+    return topics_array;
+}
+
+function get_topic_row(topic_key) {
+    return $("#" + $.escapeSelector(topic_key));
```

Comment:
Hmm, I'm not sure this is a good naming scheme; we should at least have an example in a comment.  Is this `escape("stream_id:topic_id")`?

Reply:
Yes, I will try to make it more clear.